tests/integration/graph/test_graph_basics.py

```diff
@@ -135,3 +136,34 @@ def update_graph(value):
     dash_dcc.wait_for_element('#my-graph .main-svg')
 
     assert not dash_dcc.get_logs()
+
+
+@pytest.mark.DCC675
+def test_grbs004_graph_loading_state_updates(dash_dcc):
+    app = dash.Dash(__name__, suppress_callback_exceptions=True)
+    app.layout = html.Div([
+        html.H1('loading state updates'),
+        dcc.Interval(id='interval'),
+        dcc.Graph(id='my-graph'),
+    ])
+
+    @app.callback(Output('my-graph', 'figure'), [Input('interval', 'n_intervals')])
+    def update_graph(value):
+        values = [random.randint(0, 20), random.randint(0, 20)]
+        ranges = [random.randint(0, 20), random.randint(0, 20)]
+        sleep(0.25)
```

Comment:
@ohaibbq Thanks for the test! Two comments about it though:
1. I know you aren't depending on the random numbers at all, but it still makes me uneasy to put random numbers in a test. Just use the callback input to make unique values if needed.
2. We should check both the loading state and the finished state, and do so without an explicit `sleep` time - this is both slow and potentially flaky, as timing can be erratic particularly on CI. Instead, use a `multiprocessing.Lock` pattern like we do in test `ldcp001`:

https://github.com/plotly/dash-core-components/blob/b7d1ebee327cf1b1938569a07cb5bf0dae4ecc54/tests/integration/loading/test_loading_component.py#L18-L29

If it's easier, you can test the finished state by using `dash_dcc.wait_for_no_elements(selector)` rather than finding a selector that matches only with `data-dash-is-loading` removed.

Reply:
Thanks! Its been a few years since doing selenium testing, I'll update to the Lock pattern and assert that it updates appropriately.